geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZInterStoreIntegrationTest.java

```diff
@@ -0,0 +1,696 @@
+    return scores;
+  }
+
+  private Map<String, Double> buildMapOfMembersAndScores(int start, int end) {
+    Map<String, Double> scores = new LinkedHashMap<>();
+    Random random = new Random();
+
+    for (int i = start; i <= end; i++) {
+      scores.put("player" + i, random.nextDouble());
+    }
+
+    return scores;
+  }
+
+  private Map<String, Double> buildMapOfMembersAndIdenticalScores(int start, int end,
+      double score) {
+    Map<String, Double> scores = new LinkedHashMap<>();
+
+    for (int i = start; i <= end; i++) {
+      scores.put("player" + i, score);
+    }
+
+    return scores;
+  }
+
+  Tuple tupleSumOfScores(String memberName, Map<String, Double> scores1,
+      Map<String, Double> scores2, Map<String, Double> scores3) {
+    return tupleSumOfScoresWithWeights(memberName, scores1, scores2, scores3, 1, 1, 1);
+  }
+
+  Tuple tupleSumOfScoresWithWeights(String memberName, Map<String, Double> scores1,
+      Map<String, Double> scores2, Map<String, Double> scores3, double weight1, double weight2,
+      double weight3) {
+    return new Tuple(memberName, scores1.get(memberName) * weight1
+        + scores2.get(memberName) * weight2
+        + scores3.get(memberName) * weight3);
+  }
+
+  Tuple tupleMaxOfScores(String memberName, Map<String, Double> scores1,
+      Map<String, Double> scores2, Map<String, Double> scores3) {
+    return tupleMaxOfScoresWithWeights(memberName, scores1, scores2, scores3, 1, 1, 1);
+  }
+
+  Tuple tupleMaxOfScoresWithWeights(String memberName, Map<String, Double> scores1,
+      Map<String, Double> scores2, Map<String, Double> scores3, double weight1, double weight2,
+      double weight3) {
+    return new Tuple(memberName, max(max(scores1.get(memberName) * weight1,
+        scores2.get(memberName) * weight2), scores3.get(memberName) * weight3));
+  }
+
+  Tuple tupleMinOfScores(String memberName, Map<String, Double> scores1,
+      Map<String, Double> scores2, Map<String, Double> scores3) {
+    return tupleMinOfScoresWithWeights(memberName, scores1, scores2, scores3, 1, 1, 1);
+  }
+
+  Tuple tupleMinOfScoresWithWeights(String memberName, Map<String, Double> scores1,
+      Map<String, Double> scores2, Map<String, Double> scores3, double weight1, double weight2,
+      double weight3) {
+    return new Tuple(memberName, min(min(scores1.get(memberName) * weight1,
+        scores2.get(memberName) * weight2), scores3.get(memberName) * weight3));
+  }
+
+  private Set<Tuple> convertToTuples(Map<String, Double> map,
+      BiFunction<Integer, Double, Double> function) {
+    Set<Tuple> tuples = new LinkedHashSet<>();
+    int x = 0;
+    for (Map.Entry<String, Double> e : map.entrySet()) {
+      tuples.add(new Tuple(e.getKey().getBytes(), function.apply(x++, e.getValue())));
+    }
+
+    return tuples;
+  }
+
+  private void assertThatActualScoresAreVeryCloseToExpectedScores(
+      Set<Tuple> expectedResults, Set<Tuple> results) {
+    for (Tuple expectedResult : expectedResults) {
```

Comment:
I think this method should also be testing whether all members in `expectedResults` are also exactly what is in `results`. As it is now, passing in `[a, b, c]` and `[d, e, f]` as members would never cause the assertion to be checked.